search-ui/src/pages/Details/Details.js

```diff
@@ -34,37 +38,116 @@ export default function Details() {
 
   }, [id]);
 
+  useEffect(() => {
+    setHighlight(null);
+  }, [q]);
+
+  function GetTagsHTML(tags) {
+
+    if (!!tags) {
+      let tagsHtml = tags.map((tagValue, index) => {
+        if (index < 10) {
+
+          if (tagValue.length > 30) { // check tag name length
+            // create substring of tag name length if too long
+            tagValue = tagValue.substring(0, 30);
+          }
+
+          return <button key={index} className="tag" onClick={() => setQ(tagValue)}>{tagValue}</button>;
+        } else {
+          return null;
+        }
+      });
+
+      return tagsHtml;
+    }
+
+    return null;
+  }
+
+  let tags = GetTagsHTML(document.keyPhrases);
+
+  function GetSnippets(q, content) {
+    if (!!content && q.trim() !== "") {
+      var regex = new RegExp(q, 'gi');
+
+      let matches = content.match(regex);
+
+      return matches.map((value, index) => {
+        var startIdx;
+        var ln = 400;
+
+        if (value.length > 150) {
+          startIdx = content.indexOf(value);
+          ln = value.length;
+        }
+        else {
+          if (content.indexOf(value) < 200) {
+            startIdx = 0;
+          }
+          else {
+            startIdx = content.indexOf(value) - 200;
+          }
+
+          ln = 400 + value.length;
+        }
+
+        var reference = content.substr(startIdx, ln);
```

Comment:
Now you've used slice, substring, and substr!  :)  The whole trifecta.  LOL.  Consider consistency here, just for clarity to the developers who have to use this after us.

Reply:
Thanks for catching this; standardizing on slice